
# HG changeset patch
# User Jesus Cea <jcea@jcea.es>
# Date 1316444898 -7200
# Node ID 7a41855b6196f4907ff624b303e70ca5c2a252ff
# Parent  5deecc04b7a27b3667bf9818779e5dfb0bbf94ec
Close #13007: whichdb should recognize gdbm 1.9 magic numbers
Upstream: yes

diff --git a/Lib/dbm/__init__.py b/Lib/dbm/__init__.py
--- a/Lib/dbm/__init__.py
+++ b/Lib/dbm/__init__.py
@@ -166,7 +166,7 @@ def whichdb(filename):
         return ""
 
     # Check for GNU dbm
-    if magic == 0x13579ace:
+    if magic in (0x13579ace, 0x13579acd, 0x13579acf):
         return "dbm.gnu"
 
     # Later versions of Berkeley db hash file have a 12-byte pad in

